src/NClient/NClient.Abstractions/Building/Factory/INClientFactoryOptionalBuilder.cs

```diff
@@ -14,6 +15,17 @@ namespace NClient
     /// <typeparam name="TResponse">The type of response that is used in the transport implementation.</typeparam>
     public interface INClientFactoryOptionalBuilder<TRequest, TResponse>
     {
+        #region Authorization
+        
+        /// <summary>Sets tokens for authorization.</summary>
+        /// <param name="tokens">The tokens for client authorization.</param>
+        INClientFactoryOptionalBuilder<TRequest, TResponse> WithTokenAuthorization(ITokens tokens);
```

Comment:
The method must have an IAuthorizationProvider type parameter instead of ITokens